pkgs/build-support/buildenv/builder.pl

```diff
@@ -15,7 +15,8 @@
 my $out = $ENV{"out"};
 my $extraPrefix = $ENV{"extraPrefix"};
 
-my @pathsToLink = split ' ', $ENV{"pathsToLink"};
+my @pathsToLink = split /(?<!\\) /, $ENV{"pathsToLink"};
+@pathsToLink = map { s/\\ / /g; $_ } @pathsToLink;
```

Comment:
Here's the JSON-based workaround:

```suggestion
my @pathsToLink = @{decode_json $ENV{"pathsToLinkJSON"}};
```